src/shogun/evaluation/StratifiedCrossValidationSplitting.cpp

```diff
@@ -124,7 +124,7 @@ void CStratifiedCrossValidationSplitting::build_subsets()
 				label_indices.get_element(i);
 
 		// external random state important for threads
-		current->shuffle(m_rng);
+		current->shuffle();
```

Comment:
in this case you can remove `m_rng` from `CStratifiedCrossValidationSplitting` as this is the only place it is being used in the class.